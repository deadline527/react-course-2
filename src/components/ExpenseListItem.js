import React from 'react';
import { Link } from 'react-router-dom'
import moment from 'moment';

const ExpenseListItem = ({ description, amount, createdAt, id}) => (
    <div>
        <h3><Link to={`/edit/${id}`}>{description}</Link></h3>
        <p>${ (amount / 100).toString() } - {moment(createdAt).format('MMM Do, YYYY')}</p>
    </div>
)

export default ExpenseListItem;