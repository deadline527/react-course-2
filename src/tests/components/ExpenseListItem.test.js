import React from 'react';
import { shallow } from 'enzyme'
import ExpenseListItem from '../../components/ExpenseListItem';
import toJSON from 'enzyme-to-json';
import expenses from '../fixtures/expenses'

test('Should render ExpenseListItem with expenses', () => {
    const wrapper = shallow(<ExpenseListItem {...expenses[0]} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
})

// test('Should render ExpenseListItem without expenses', () => {
//     const wrapper = shallow(<ExpenseListItem {...expenses[4]} />);
//     expect(toJSON(wrapper)).toMatchSnapshot();
// })

