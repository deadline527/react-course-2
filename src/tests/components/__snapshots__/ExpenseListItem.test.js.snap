// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpenseListItem with expenses 1`] = `
<div>
  <h3>
    <Link
      replace={false}
      to="/edit/1"
    >
      gum
    </Link>
  </h3>
  <p>
    $
    1.95
     - 
    Dec 31st, 1969
  </p>
</div>
`;
